COBOL Programming Course #1 - Getting Started/COBOL Programming Course #1 - Getting Started.md

```diff
@@ -1903,6 +1903,308 @@ Notice that this line tells you to focus on the GROSS-PAY picture clause in orde
 
 \newpage
 
+# Tables handling
+
+This section introduces the concept of tables, which are a collection of data items that have the same description. The subordinate items are called table elements. A table is the COBOL equivalent of arrays.
+
+The objective of this chapter is to provide information for the reader to be able to handle tables inside COBOL programs.
+
+## Defining a table
+
+To code a table, we need to give the table a group name and define a subordinate item which we are repeating n times.
+
+```
+01  TABLE-NAME.
+    05  SUBORDINATE-NAME OCCURS n TIMES.
+        10  ELEMENT1  PIC X(2).
+        10  ELEMENT2  PIC 9(2).
+```
+
+In the example above, TABLE-NAME is the name of the group item. The table also contain a subordinate item called SUBORDINATE-NAME which we are repeating n times. Each of the SUBORDINATE-ITEM have 2 elementary items, ELEMENT1 and ELEMENT2. In this case, we called SUBORDINATE-NAME as the table element definition (since it includes the OCCURS clause). Note that the OCCURS clause cannot be used in a level-01 description.
+
+Alternatively, we can also make simpler tables:
+
+```
+01  TABLE-NAME.
+    05  SUBORDINATE OCCURS n TIMES    PIC X(10).
+```
+
+In this case, TABLE-NAME contains n SUBORDINATE item, each can contain up to 10 alphanumeric characters.
+
+We can also nest multiple OCCURS elements to create table of additional dimensions, up to a limit of seven dimensions. Note the example below:
+
+```
+01  PROGRAM-DETAILS.
+    05  PROGRAM-DEGREE                PIC X(32).
+    05  COURSE-DETAILS OCCURS 10 TIMES.
+        10  COURSE-NAME               PIC X(32).
+        10  INSTRUCTOR-ID             PIC 9(10).
+        10  ASSIGNMENT-DETAILS OCCURS 8 TIMES.
+            15  ASSIGNMENT-NAME       PIC X(32).
+            15  ASSIGMMENT-WEIGHTAGE  PIC 9(03).
+```
+
+Here, we are defining a degree program which have 10 course and each course will have 8 assignments. What if we don't know beforehand how many times will a table element occurs? To solve that, we can use variable-length table, using the OCCURS DEPENDING ON (ODO) clause which we will be going into more details on a later section.
```

Comment:
`degree program which has 10 courses and ...`

`What if we don't know how many times a table element will occur?`